examples/systemcalls/systemcalls.c

```diff
@@ -85,15 +127,73 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     command[count] = command[count];
 
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
 
-    va_end(args);
+    // This solution is rewritten and expanded version of the code suggested to us https://stackoverflow.com/a/13784315/1446624
 
+    int childPid;
+     openlog("assignment3-part1",LOG_PID, LOG_USER); 
```

Comment:
@Nicholas-Buckley77 think about how you could avoid duplicating logic between these two functions